### PR TITLE
doc/source/hacking: Document wheel building infrastructure

### PR DESCRIPTION
**Pull request by ghost**
There is a lot to remember in terms of wheel building, this is an
attempt to write it all down before I forget.

### CODE REVIEW

**Comment by jjardon** [comment]
This looks good to me: @gtristan @abderrahim @juergbi  ok to merge?

File: doc/source/hacking/updating_python_deps.rst [hunk 1]
```diff
@@ -20,3 +20,89 @@ changing the ``.in`` file, run the following to update the matching ``.txt``
...
+~~~~~~~
+
+The ``tox.ini`` file defines the environments where the BuildStream test suite
+runs.  Every ``py{3.x,3.y}`` list must be updated to contain the new version
+number such as ``311`` for CPython 3.11.
+
+Use ``tox -e py311-nocover`` to run the test suite with the new version of
+Python.
+
+pyproject.toml
```
Comment (gtristan) [inline doc/source/hacking/updating_python_deps.rst:40]: Good addition to the hacking !

note that here we also have to bump the minimum required version of cython which includes support for the new cpython version.

Comment (ghost) [reply doc/source/hacking/updating_python_deps.rst:40]: i see, and i assume this is a change request so will push an update when I have a chance

Comment (gtristan) [reply doc/source/hacking/updating_python_deps.rst:40]: I didn’t notice how old this was, don’t worry I’ll do it :)

**Pull request merged**